mujoco_py/mjpid.pyx

```diff
@@ -169,13 +169,18 @@ cdef mjtNum c_pi_cascade_bias(const mjModel*m, const mjData*d, int id):
     """
     cdef double dt_in_sec = m.opt.timestep
     cdef int NGAIN = int(const.NGAIN)
+    cdef double smooth_pos_setpoint;
 
     cdef double Kp_cas = m.actuator_gainprm[id * NGAIN + IDX_CAS_PROPORTIONAL_GAIN]
 
-    # Apply Exponential Moving Average smoothing to the position setpoint
-    ctrl_ema = d.userdata[id * NUM_USER_DATA_PER_ACT + IDX_CAS_STORED_EMA_SMOOTH]
-    smoothing_factor = m.actuator_gainprm[id * NGAIN + IDX_CAS_EMA_SMOOTH]
-    smooth_pos_setpoint = (smoothing_factor * ctrl_ema) + (1 - smoothing_factor) * d.ctrl[id]
+    # Apply Exponential Moving Average smoothing to the position setpoint, applying warmstart
+    # on the first iteration.
+    if d.time > 0.0:
```

Comment:
Is this the best condition to check for? Ie: it happens at d.time == 0, but could it also happen under certain resets that clear data, but do not reset t?

Reply:
hmm I'm not sure if we reset the PID controller ever. are you aware of an action triggering that?